plugins/outputs/amqp/amqp.go

```diff
@@ -260,6 +260,9 @@ func (q *AMQP) Write(metrics []telegraf.Metric) error {
 					return err
 				}
 			} else {
+				if err := q.client.Close(); err != nil {
+					q.Log.Errorf("Closing connection failed: %v", err)
+				}
 				q.client = nil
```

Comment:
Thanks for this - the GitHub diff looks weird, but this does look good to close the connection on the 2nd and additional publish attempts.